Update CAA docs

davidfischer opened the pull request:
- Note that CAA records are usually at the root
- They don't *have* to be at the root but you can't have a CNAME and CAA record for the same subdomain

humitos commented on docs/custom_domains.rst:
```diff
@@ -51,10 +51,11 @@ You can also host your documentation from your own domain.
+         CAA records are typically configured at the root domain.
          See the `Cloudflare CAA FAQ`_ for details.
-         We need a record that looks like this: ``0 issue "digicert.com"`` in response to ``dig +short CAA <domain>``
+         We need a record that looks like this: ``0 issue "digicert.com"`` in response to ``dig +short CAA <rootdomain>``
```
Does it worth to put an example here? Like, if your domain is `docs.example.org`, your root domain is `example.org`.

davidfischer replied:
I reworked this section and added an example.

davidfischer commented:
I updated the docs to add a troubleshooting section where I moved the CAA notes, the Cloudflare-to-Cloudflare notes, and discuss delays, retrying, and avoiding downtime.

ericholscher commented on docs/custom_domains.rst:
```diff
@@ -112,6 +81,68 @@ for our own domains including ``*.readthedocs.io``, ``*.readthedocs-hosted.com``
+You will need to update your CAA records to allow us to issue the certificate.
+
+* For |org_brand|, we use Cloudflare which uses Digicert as a CA. See the `Cloudflare CAA FAQ`_ for details.
+* For |com_brand|, we use AWS Certificate Manager as a CA. See the `Amazon CAA guide`_ for details.
```
I do think an example of what a "proper" record looks like for confirmation would be useful, but maybe a bit much.

davidfischer replied:
There's a community example a few lines up (palletsprojects.com). Since this is outside the tab, it's not as nice to provide an example .com and a .org example.
